src/AllUserSleep.js

```diff
@@ -1,7 +1,87 @@
+var dayjs = require("dayjs");
+var duration = require('dayjs/plugin/duration')
+dayjs.extend(duration);
+var isBetween = require('dayjs/plugin/isBetween')
+dayjs.extend(isBetween)
+
 class AllUserSleep {
   constructor(sleepData) {
     this.sleepData = sleepData;
   }
+
+  calcAvgSleepQuality(sleepData) {
+    const totalSleepQuality = sleepData.reduce((total, num) => {
```

Comment:
This may be a little nit-picky- I'm curious about the use  `sleepData` vs `this.sleepData` here!! I would advocate for `this.sleepData`- let's say in the future you decide to modify `this.sleepData` in the constructor to be an object like below, where each `key` is a user's `id`...

`this.sleepData = {
     1: [ 
          {...data...},
          {...data...},
          {...data...},
     ],
     2: [
          {...data...},
          {...data...},
     ],
    ...
}
`

You would want your implementation code to reference `this.sleepData` rather than your original data passed in as a prop!